### PR TITLE
28 setup doxygen doc

### LINKED ISSUE #28: Setup doxygen doc
- create Doxyfile
- add submodule doxygen-awesome for nice formatting
- create github action for automatic deployment

### PR DESCRIPTION
**Pull request by gwolf-011235**
#### Create Doxyfile
- is used to create doxygen documentation
- A template Doxyfile can be generated by doxygen -g
- To show difference between template and ours use doxygen -x doc/Doxyfile
- Landing page is README.md of repo
- Doxygen version in dev container is 1.9.4 (bookworm): [package](https://packages.debian.org/bookworm/doxygen)

#### Add submodule [doxygen-awesome-css](https://github.com/jothepro/doxygen-awesome-css/tree/main) 
- is used to create more stylish documentation; is fixed at latest stable release
- is referenced in Doxyfile as HTML_EXTRA_STYLESHEET and HTML_EXTRA_FILES
- changes the design of documentation
- also adds functionality (darkmode, fragment copy button, paragraph link) which are activated in doc/header.html

#### Implements github action [Doxygen GitHub Pages Deploy Action](https://github.com/DenverCoder1/doxygen-github-pages-action) 
- automatically create doxygen documentation and deploy to branch GH-pages
- is executed by push on main
- executes on ubuntu latest (noble). This has doxygen version 1.9.8 - [package](https://packages.ubuntu.com/noble/doxygen). This should not be a problem since backwards compatible. If it is dev container will be updated.

#### Add package graphviz to dev container 
- graphviz is used by doxygen to create svg graphs
- documentation can be created as a test in dev container

#### Delete doubled main
- in the project root directory was an old copy of main.cpp -> deleted

Closes #28

### CODE REVIEW

**Review by QCHR1581 — APPROVED** [review]
Great work. Tested and works as intended. Only trouble I had was the not initialised submodule but that can be solved by using the command: git submodule update --init.

File: .github/workflows/doxygen-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: DenverCoder1/doxygen-github-pages-action@v1.3.1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          config_file: doc/Doxyfile_bonus
```
Comment (u413-284-si) [inline .github/workflows/doxygen-gh-pages.yml:16]: The Doxyfile doesn't have the _bonus suffix

**Review by u413-284-si — CHANGES_REQUESTED** [review]
Just one small comment.

File: .github/workflows/doxygen-gh-pages.yml (hunk 1 above)
Comment (gwolf-011235) [reply .github/workflows/doxygen-gh-pages.yml]: Nice catch - removed suffix

**Review by u413-284-si — APPROVED** [review]

**Pull request merged**